pandas/core/arrays/interval.py

```diff
@@ -192,6 +199,10 @@ class IntervalArray(IntervalMixin, ExtensionArray):
     can_hold_na = True
     _na_value = _fill_value = np.nan
 
+    _dtype: IntervalDtype
+    _left: np.ndarray | DatetimeArray | TimedeltaArray
+    _right: np.ndarray | DatetimeArray | TimedeltaArray
```

Comment:
@simonjayhawkins any idea how to annotate "_left and _right have the same type"?

Reply:
could get rid of a few ignores with...

```diff
diff --git a/pandas/core/arrays/interval.py b/pandas/core/arrays/interval.py
index 1f7c5ebb04..39a1a8e720 100644
--- a/pandas/core/arrays/interval.py
+++ b/pandas/core/arrays/interval.py
@@ -8,6 +8,7 @@ from operator import (
 import textwrap
 from typing import (
     TYPE_CHECKING,
+    Generic,
     Sequence,
     TypeVar,
     cast,
@@ -94,6 +95,8 @@ if TYPE_CHECKING:
 
 IntervalArrayT = TypeVar("IntervalArrayT", bound="IntervalArray")
 
+S = TypeVar("S", np.ndarray, DatetimeArray, TimedeltaArray)
+
 _interval_shared_docs: dict[str, str] = {}
 
 _shared_docs_kwargs = {
@@ -194,14 +197,14 @@ for more.
         ),
     }
 )
-class IntervalArray(IntervalMixin, ExtensionArray):
+class IntervalArray(IntervalMixin, ExtensionArray, Generic[S]):
     ndim = 1
     can_hold_na = True
     _na_value = _fill_value = np.nan
 
     _dtype: IntervalDtype
-    _left: np.ndarray | DatetimeArray | TimedeltaArray
-    _right: np.ndarray | DatetimeArray | TimedeltaArray
+    _left: S
+    _right: S
 
     # ---------------------------------------------------------------------
     # Constructors
```